src/pages/components/CertificateForm.js

```diff
@@ -123,21 +125,53 @@ export default function CertificateForm(props) {
       callApiFnChain,
       callApiFnFinal,
       update,
-      navigate,
+      current,
     ]
   )
 
+  const cert = dataFinal?.state === 'initiated' ? rest[current] : {}
+  React.useEffect(() => {
+    const get = async () => {
+      const url = `${persona.origin}/v1/certificate/${dataChain.original_token_id}`
```

Comment:
From my understanding we will only get the `original_token_id` later isn't it and if that is the case would this need a check first?

Reply:
so you have your watcher and it does this `while` look until we get, then you put it in `dataChain` hook so it does contain, this is why there is a check below 

```
    if (dataChain?.original_token_id) get()
```